packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -510,32 +549,69 @@ export class GraphiQL extends React.Component<GraphiQLProps, GraphiQLState> {
                 readOnly={this.props.readOnly}
               />
               <section
-                className="variable-editor"
-                style={variableStyle}
+                className="extra-editor"
+                style={extraEditorStyle}
                 aria-label="Query Variables">
                 <div
-                  className="variable-editor-title"
-                  id="variable-editor-title"
+                  className="extra-editor-title"
+                  id="extra-editor-title"
                   style={{
-                    cursor: variableOpen ? 'row-resize' : 'n-resize',
+                    cursor: extraEditorOpen ? 'row-resize' : 'n-resize',
                   }}
-                  onMouseDown={this.handleVariableResizeStart}>
-                  {'Query Variables'}
+                  onMouseDown={this.handleExtraEditorResizeStart}>
+                  <div
+                    style={{
+                      cursor: 'pointer',
+                      color: variableEditorActive ? 'red' : '#000',
+                      display: 'inline-block',
+                    }}
+                    onClick={this.handleOpenVariableEditorTab}
+                    onMouseDown={this.handleTabClickPropogation}>
+                    {'Query Variables'}
+                  </div>
+                  <div
+                    style={{
+                      cursor: 'pointer',
+                      color: headerEditorActive ? 'red' : '#000',
+                      display: 'inline-block',
+                      marginLeft: '20px',
+                    }}
+                    onClick={this.handleOpenHeaderEditorTab}
+                    onMouseDown={this.handleTabClickPropogation}>
+                    {'Query Headers'}
```

Comment:
This might be nitpicking, but `Query Headers` feels a bit confusing to me. I think `Request Headers` may be more accurate, as the headers are part of the transport mechanism, not the graphql spec itself, if that makes sense.

Reply:
agreed!

Reply:
Updated 👍